db/blob_file_state.cc

```diff
@@ -0,0 +1,141 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "db/blob_file_state.h"
+
+#include "logging/event_logger.h"
+#include "rocksdb/slice.h"
+#include "rocksdb/status.h"
+#include "test_util/sync_point.h"
+#include "util/coding.h"
+
+#include <ostream>
+#include <sstream>
+
+namespace rocksdb {
+
+namespace {
+
+// Tags for custom fields. Note that these get persisted in the manifest,
+// so existing tags should not be modified.
+enum CustomFieldTags : uint32_t {
+  kEndMarker,
+
+  // Add forward compatible fields here
+
+  /////////////////////////////////////////////////////////////////////
+
+  kForwardIncompatibleMask = 1 << 6,
+
+  // Add forward incompatible fields here
+};
+
+}  // anonymous namespace
+
+void BlobFileState::EncodeTo(std::string* output) const {
```

Comment:
I wonder whether returning a boolean is better, assuming sometime in the future we want to notify the caller of success or failure. Note that `VersionEdit::EncodeTo` returns boolean.

Reply:
I'm actually thinking about changing it to `Status` to make it more flexible; there might be more than one failure case.

Reply:
Actually, I'd say let's leave this for now; we can always change the signature later if it becomes necessary. I would also argue that `VersionEdit::EncodeTo` returning a `bool` is not a good design: it only returns `false` if a newly added file does not have a valid smallest or largest key, which could have (should have) been detected when the file was added to the edit.

Reply:
It does not sound like an error in "encoding", maybe we should change VersionEdit::EncodeTo.
OK with your approach here.